### PR TITLE
refactor: rewrite python provider healthcheck in Lua

### PR DESCRIPTION
**Pull request by dundargoc**
This is required to remove the vimscript checkhealth functions.

### CODE REVIEW

File: runtime/lua/provider/health.lua [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  vim.fn.system(cmd)
+  return vim.v.shell_error == 0
+end
+
+local function executable(exe)
+  return vim.fn.executable(exe) == 1
+end
+
+local function empty(s)
+  return vim.trim(s) == ''
```
Comment (echasnovski) [inline runtime/lua/provider/health.lua]: I'd use `s:find('^%s*$') ~= nil` as is. If inside wrapper function, `is_blank()` is better: "empty" <-> `""`, blank <-> "consists from whitespace".

File: runtime/lua/provider/health.lua [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+local function clipboard()
+  vim.health.report_start('Clipboard (optional)')
+
+  if
+    os.getenv('TMUX')
+    and executable('tmux')
+    and executable('pbpaste')
+    and not cmd_ok('pbpaste')
+  then
+    local tmux_version = vim.fn.matchstr(vim.fn.system('tmux -V'), [[\d\+\.\d\+]])
```
Comment (echasnovski) [inline runtime/lua/provider/health.lua:152]: This seems to be equivalent to `string.match(vim.fn.system('tmux -V'), '%d+%.%d+')`.

Comment (dundargoc) [reply runtime/lua/provider/health.lua:152]: You're right. Thanks!

File: runtime/lua/provider/health.lua (hunk 1 above)
Comment (dundargoc) [reply runtime/lua/provider/health.lua]: Good call on the "is_blank". What's the benefit of using `:find` compared to vim.trim?

Comment (clason) [reply runtime/lua/provider/health.lua]: `find` is faster than calling `match` and checking the output.

Comment (dundargoc) [reply runtime/lua/provider/health.lua]: That's good enough for me :)

**Pull request merged**